src/buffer.c

```diff
@@ -2045,25 +2045,6 @@ set_buffer_if_live (Lisp_Object buffer)
     set_buffer_internal (XBUFFER (buffer));
 }
 
-DEFUN ("barf-if-buffer-read-only", Fbarf_if_buffer_read_only,
-				   Sbarf_if_buffer_read_only, 0, 1, 0,
-       doc: /* Signal a `buffer-read-only' error if the current buffer is read-only.
-If the text under POSITION (which defaults to point) has the
-`inhibit-read-only' text property set, the error will not be raised.  */)
-  (Lisp_Object position)
-{
-  if (NILP (position))
-    XSETFASTINT (position, PT);
```

Comment:
I missed that this is setting position not just returning it.

Reply:
Hmmm. I don't see how this is setting point. It looks like this is setting the argument position to point ? As I see it, the function's purpose is only to signal an error if buffer/position is read only.

Reply:
It isn't. never mind. Your code mimics the intent just fine.